source/base/utilities.cc

```diff
@@ -484,6 +484,22 @@ namespace Utilities
 
 
 
+  template <typename Number>
+  Number
+  cutoff_digits(const Number number, const unsigned int n_digits)
+  {
+    AssertThrow(n_digits >= 1, ExcMessage("invalid parameter."));
+
+    const int order = std::floor(std::log10(std::fabs(number)));
+
+    const Number factor = std::pow(10.0, (Number)(-order + int(n_digits) - 1));
```

Comment:
We typically spell out double <-> integer conversions like this:
```suggestion
    const Number factor = std::pow(10.0, (Number)(-order + static_casts<int>(n_digits) - 1));
```